projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_error_handler.py

```diff
@@ -228,6 +228,12 @@ def _handle_metadata_exception_with_backoff(
                 classname_with_package="impala.error.OperationalError",
                 exception_message_matcher_regex=".*ImpalaRuntimeException: Error making 'updateTableColumnStatistics'.*",
             )
+            or errors.exception_matches(
+                exception,
+                classname_with_package="impala.error.HiveServer2Error",
+                exception_message_matcher_regex="AuthorizationException: User .*does not have privileges to "
```

Comment:
The issue with Impala sometimes taking time to propagate new authorization settings and rasing AuthorizationException is not related to invalidate metadata /refresh . 

but to any query. So why not re-try on any query?

I

Reply:
Refactored, as discussed offline

Reply:
It was agreed that the previous change to single out `Authorization errors` on refresh/invalidate metadata and handle them, was too limiting, and instead a different approach is needed. The refresh/invalidate metadata queries are issued by the error handling logic, not by the user, therefore we need to swallow these errors, and just retry the initial query. This way, if the initial query fails again, it would be somewhat expected, and the log message for the error would make more sense to the user, instead of receiving a notification for an error related to refresh/invalidate metadata, which would not make any sense.